### DISCUSSION TITLE
RPi Pico - what UART is used by USB

### DISCUSSION THREAD

**Discussion by SteKme** [micropython/micropython#11080] [Using MicroPython] [answered]
Hi,

I tried hard, but couldn't find, unfortunately:

-- please what UART is used by the USB port of RPi Pico? There are UART0+1 available on pins of Pico, but the USB connector provides another serial communication with the PC (in uPython, it is acessed as stdin, stdout, indeed). 
Is the USB serial communication on top of the two uarts, or when the USB data cable is plugged in, one of the two uarts is used for this purpose and therefore unavailable for other purposes?

Thank You so much,
Stepan

**Comment by robert-hh** [comment, marked as answer]
USB and UART are independent. So you can use both UART0 and UART1 for other tasks.
If you like to, you can have stdin/stdout at UART in addition to USB in the standard build, using dupterm. Or you can create a special build with stdin/out at UART, and even disable USB.
